### DISCUSSION TITLE
Pull model of tasks

### DISCUSSION THREAD

**Discussion by pavetok** [go-task/task#1054] [General]
Hey, thanks for creating a great tool!

I'm trying to implement pull-like model with Task.  E.g. I have `image` task, which have `binary` task as dependency. `image` status checks existence with `docker inspect`. So, I expect that `binary` will be skipped at all when image already exist.

It seems Task based on neither pull or push model? In a push model changed deps should triger dependent task despite its status. It's like bottom-up propagation of changes. In a pull model failed status should trigger task deps or should trigger nothing otherwise. It's like top-down propagation of demand. So, it seems Task based on some third model (let's call it independency model)? Nor zero status skips deps, nor changed deps trigers dependent tasks. Each task does its work based solely on its own status. Is it feasible to implement all three models?

Related push model discussion https://github.com/go-task/task/discussions/669

Related pull model discussion https://github.com/go-task/task/discussions/787

**Comment by andreynering** [comment]
Hi @pavetok,

You're right, each task is evaluated independently of the others.

This seems to be good enough for most users, although sometimes people asks for improvements in this area (dependencies invalidating parent tasks, etc).

An option to choose how it'd work would be interesting for users, as some users want a different behavior here. The drawback would be having to maintain ~3 different models in the code, which could be a bit tricky.

This may happen eventually, but it's not on the roadmap at the moment. If you're really interested into this feature, you can consider creating an issue with more detail on how you see this working.
